data/data_interface.py

```diff
@@ -0,0 +1,160 @@
+import pandas as pd
+from typing import Sequence, Any, Tuple
+import os
+import pathlib
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import StandardScaler, MinMaxScaler
+from synthetic_data import create_synthetic_data
+import numpy as np
+
+class DataInterface():
+    def __init__(self, data : pd.DataFrame, file_path : str, continuous_features : Sequence[str],
+                ordinal_features : Sequence[str], categorical_features : Sequence[str], immutable_features : Sequence[str],
+                target_feature : str, target_mapping : Any = None, scaling_method : str = "MinMax", encoding_method : str = "OneHot"):
+        """
+        Creates a data interface with the specified, continuous features, ordinal features, and categorical features.
+
+        Features can be normalized, standardized, or encoded as desired. Additionally data can be split into train and test data.
+
+        Parameters
+        ----------
+        data : pd.DataFrame
+            A dataframe of the data to load. If None, we will attempt to load using file_path
+        file_path : str
+            The path to load your data from. Not required with data is not None.
+        continuous_features : Sequence[str]
+            A sequence of the continous features.
+        ordinal_features : Sequence[str]
+            A sequence of the ordinal features.
+        categorical_features : Sequence[str]
+            A sequence of the categorical/nominal features.
+        immutable_features : Sequence[str]
+            A sequence of the immutable features.
+        target_feature : str
+            The column name of the target feature
+        target_mapping : Any
+            A mapping of the target features to some values
+        scaling_method : str
+            String indicating how you with to scale your data. Options are MaxMin normalization and standardization 
+        encoding_method : str
+            Encoding method for categorical features. Only one-hot for now.
+        """
+        self._continuous_features = continuous_features
+        self._ordinal_features =  ordinal_features
+        self._categorical_features = categorical_features
+        self._immutable_features = immutable_features
+        self._target_feature = target_feature
+        self._target_mapping = target_mapping
```

Comment:
Might be helpful to include in the docstring that these are access-only and have getter methods (i.e. are intended to be immutable but accessible from within step_lib.py).